src/components/encounter/ohri-encounter-form.component.tsx

```diff
@@ -544,6 +553,42 @@ export const OHRIEncounterForm: React.FC<OHRIEncounterFormProps> = ({
         return Promise.all(saveAttachmentPromises).then(() => response);
       });
     }
+
+    if (encounterForSubmission.obs?.length || encounterForSubmission.orders?.length) {
+      if (Object.keys(encounterContext?.patientIdentifier).length) {
+        return saveIdentifier(patient, encounterContext.patientIdentifier)
+          .then(() => saveEncounterWithAttachments(encounterForSubmission))
+          .catch((error) => {
+            showSnackbar({
+              title: t('errorDescriptionTitle', 'Error on saving form'),
+              subtitle: t('errorDescription', error),
+              kind: 'error',
+              isLowContrast: false,
+            });
+          });
+      } else {
+        return saveEncounterWithAttachments(encounterForSubmission);
+      }
+    }
+  };
```

Comment:
I'm confused about two things:
1. We seem to have duplicated blocks of: `if (encounterForSubmission.obs?.length || encounterForSubmission.orders?.length) {`
2. Why do we attempt saving identifiers within conditional blocks tangential to encounter?

Reply:
My bad, forgot to delete the code after refactoring